doc/quick_intro.mld

```diff
@@ -0,0 +1,79 @@
+order to generate or alter the source code. The preprocessing is thus left to
+external programs, written by the community, and specialized for their own
+tasks. However, without a unification framework, the following issues would
+arise:
+
+- Difficulty in using several preprocessors, with incompatibility or composition
+  issues,
+- Duplication of code and efforts on the different preprocessors,
+- Performance loss with many phases (parsing, pretty-printing, ...) being
+  executed multiple times.
+
+{1 Ppxlib}
+
+The goal of Ppxlib is to provide this unifying framework. It sits in between the
+OCaml compiler and toolchain, and the PPX authors, providing an API for them.
+One could sum up the Ppxlib features as:
+
+- Deal with all boilerplate such as parsing the input, output the rewritten
+  output, generate an executable, ...
+- Generate a single executable for multiple transformations, dealing itself with
+  possible composition issues,
+- Integrate well with Dune and Merlin,
+- Provide a more stable API than the compiler for manipulating the AST,
+- A single PPX codebase works on every OCaml version of the AST,
+- Define restricted rewriters whose semantic ensure better confidence for the
+  user and better compositional semantics,
+- Provide many helpers to pattern-match and generate AST nodes, traverse the
+  AST.
+
+{1 This guide}
+
+This guide is separated into several parts, whose reading needs not necessarily
+to be linear.
+
+First, we focus on the {{!driver}driver}: how it is generated, executed, and
+more importantly what it does exactly, from migrating the AST, to the different
+rewriting phases it goes through.
+
+After that, we explain {{!"writing-ppxs"}the different kinds} of transformations
+that ppxlib supports, and how to register them to the driver. This section only
+describes the transformations and their properties, not how to actually
+manipulate the AST.
+
+The part where we discuss how to manipulate the AST is split in three pages:
+{{!"generating-code"}generating AST nodes} to generate OCaml code,
+{{!"matching-code"}destructing AST nodes} to extract information and act
+differently depending on what is extracted, and {{!"ast-traversal"}traversing
+the AST} to use [fold], [iter] and [map] on the AST. This code-manipulation part
+explains both using the modules {!Ast_builder}, {!Ast_pattern} and the Ppxlib's
+PPX {!Metaquot}.
+
+We finally discuss several {{!"good-practices"}good practices}, such as how to
+properly report errors, how to test your PPX, or how to migrate from other PPX
+libraries such as [OMP] and [ppx_deriving].
+
+We end by including some {{!"examples"}examples}, which you can also find in the
+[examples] {{:https://github.com/ocaml-ppx/ppxlib/tree/main/examples}folder} of
+Ppxlib's repository.
+
+{1 PPXs and Ppxlib history}
+
+The preprocessing history of OCaml started long before Ppxlib. Several tools
+have occupied the place that Ppxlib occupies now, TODO
```

Comment:
not worth spending too much time on this, but the obvious ones are camlp4 and camlp5.  The original PPX discussions are archived here: https://lists.ocaml.org/pipermail/wg-camlp4/2013-January/000000.html

Reply:
I added some links and a link to contribute to the doc (I don't really have much more time to write more sections, I'll focus on getting this merged. Missing sections would be added later)